### ISSUE TITLE
Links in @family tag don't have anchors

### ISSUE THREAD

**Issue by llrs** [r-lib/roxygen2#1650]
CRAN (and Bioconductor) wish that all cross-references were anchored (And testing via a new environmental variable).
I was checking the current state of CRAN's cross references and I found out that some of my own cross-references were missing the anchor (See for example the [source](https://github.com/ropensci/rtweet/blob/6fc9a961d15e9c1f9efefdff9a0c821dfbbb2851/R/auth.R#L14) and the [output](https://github.com/ropensci/rtweet/blob/6fc9a961d15e9c1f9efefdff9a0c821dfbbb2851/man/auth_setup_default.Rd#L34-L-L38)). This happens when they are generated with `#' @family group`.

``` r
library(roxygen2)
roc_proc_text(rd_roclet(), "
  #' Title1
  #'
  #' See [bar()]
  #' @family check
  #' @md 
  foo <- function() {}
  
  #' Title2
  #'
  #' @family check
  bar <- function() {}
  "
)
#> $foo.Rd
#> % Generated by roxygen2: do not edit by hand
#> % Please edit documentation in ./<text>
#> \name{foo}
#> \alias{foo}
#> \title{Title1}
#> \usage{
#> foo()
#> }
#> \description{
#> See \code{\link[=bar]{bar()}}
#> }
#> \seealso{
#> Other check: 
#> \code{\link{bar}()}
#> }
#> \concept{check}
#> 
#> $bar.Rd
#> % Generated by roxygen2: do not edit by hand
#> % Please edit documentation in ./<text>
#> \name{bar}
#> \alias{bar}
#> \title{Title2}
#> \usage{
#> bar()
#> }
#> \description{
#> Title2
#> }
#> \seealso{
#> Other check: 
#> \code{\link{foo}()}
#> }
#> \concept{check}
```

<sup>Created on 2024-08-27 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

I think the root cause is that it is handled different than in other cases where links are created (as in markdown it works well), see:
https://github.com/r-lib/roxygen2/blob/9652d15221109917d46768e836eaf55e33c21633/R/rd-family.R#L56-L60

Different from:

https://github.com/r-lib/roxygen2/blob/9652d15221109917d46768e836eaf55e33c21633/R/object-import.R#L36-L41

I'm not sure how it handles multiple topics in the same file after #1109 or how to find the right topic on the rd-family.R, but fixing this will eventually help solve a note from packages that use roxygen2.

**Comment by llrs** [comment]
Apologies, I think they are not needed in this case (yet?) because the links are always to topics in the same package and CRAN doesn't complain about it. 

I am not sure if then the anchors are needed [according to the documentation](https://cran.r-project.org/doc/manuals/r-devel/R-exts.html#Cross_002dreferences). I leave it open in case this helpful or you want to harmonize both ways have links.

**Comment by hadley** [comment]
Yeah, anchors shouldn't be needed here since it's within the package.
